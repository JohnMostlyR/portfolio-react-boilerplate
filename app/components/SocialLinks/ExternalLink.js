import React from 'react';
import PropTypes from 'prop-types';

import StyledA from './StyledA';
import Description from './Description';

import IconCodepen from '../../components/IconCodepen';
import IconFreeCodeCamp from '../../components/IconFreeCodeCamp';
import IconGithub from '../../components/IconGithub';
import IconLinkedin from '../../components/IconLinkedin';
import IconExternalLink from '../../components/IconExternalLink';

export const VALID_ICONS = [
  'free-code-camp',
  'github',
  'linkedin',
  'linkedin-square',
  'codepen',
];

function ExternalLink({ color, description, href, faIcon }) {
  let renderThis = <span />;

  switch (faIcon) {
    case 'free-code-camp':
      renderThis = <IconFreeCodeCamp />;
      break;
    case 'github':
      renderThis = <IconGithub />;
      break;
    case 'linkedin':
    case 'linkedin-square':
      renderThis = <IconLinkedin />;
      break;
    case 'codepen':
      renderThis = <IconCodepen />;
      break;
    default:
      renderThis = <IconExternalLink />;
  }

  return (
    <StyledA
      href={href}
      target="_blank"
      rel="noopener noreferrer"
      color={color}
    >
      {renderThis}
      <Description>{description}</Description>
    </StyledA>
  );
}

ExternalLink.propTypes = {
  color: PropTypes.string,
  description: PropTypes.string,
  href: PropTypes.string,
  faIcon: PropTypes.oneOf(VALID_ICONS),
};

ExternalLink.defaultProps = {
  color: 'inherit',
  description: 'ExternalLink to',
  href: '#',
};

export default ExternalLink;
