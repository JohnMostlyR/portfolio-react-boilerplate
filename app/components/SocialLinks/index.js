import React from 'react';
import ReactTooltip from 'react-tooltip';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

import ExternalLink from './ExternalLink';
import StyledUL from './StyledUL';
import StyledLI from './StyledLI';
import messages from './messages';

import ThemeContext from '../../styles/theme';

const Nav = styled.nav`
  @supports (display: flex) {
    align-items: center;
    display: flex;
  }
`;

class SocialLinks extends React.PureComponent {
  render() {
    const externalLinks = [
      {
        icon: 'free-code-camp',
        name: 'freeCodeCamp',
        url: 'https://www.freecodecamp.org/mensae',
      },
      {
        icon: 'github',
        name: 'GitHub',
        url: 'https://github.com/Mensae',
      },
      {
        icon: 'linkedin',
        name: 'LinkedIn',
        url: 'https://nl.linkedin.com/in/jmeester',
      },
      {
        icon: 'codepen',
        name: 'Codepen',
        url: 'https://codepen.io/jmeester',
      },
    ];

    return (
      <Nav aria-labelledby="SocialLinks-title">
        <FormattedMessage {...messages.header}>
          {message => (
            <h1 id="SocialLinks-title" hidden aria-hidden="false">
              {message}
            </h1>
          )}
        </FormattedMessage>
        <StyledUL
          innerRef={e => {
            this.list = e;
          }}
        >
          {externalLinks.map(externalLink => (
            <StyledLI key={externalLink.url} data-tip={externalLink.name}>
              <ThemeContext.Consumer>
                {({ pageHeader }) => (
                  <ExternalLink
                    href={externalLink.url}
                    faIcon={externalLink.icon}
                    description={externalLink.name}
                    color={pageHeader.backgroundColor}
                  />
                )}
              </ThemeContext.Consumer>
            </StyledLI>
          ))}
        </StyledUL>
        <ReactTooltip />
      </Nav>
    );
  }
}

export default SocialLinks;
