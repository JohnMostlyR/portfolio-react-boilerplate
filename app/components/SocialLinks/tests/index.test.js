import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';

import SocialLinks from '../index';

import ThemeContext, { theme } from '../../../styles/theme';

describe('<SocialLinks />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = mount(
      <IntlProvider locale="en">
        <ThemeContext.Provider value={theme}>
          <SocialLinks />
        </ThemeContext.Provider>
      </IntlProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
