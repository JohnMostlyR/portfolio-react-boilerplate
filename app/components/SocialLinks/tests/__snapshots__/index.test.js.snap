// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLinks /> should render and match the snapshot 1`] = `
<IntlProvider
  locale="en"
>
  <SocialLinks>
    <SocialLinks__Nav
      aria-labelledby="SocialLinks-title"
    >
      <nav
        aria-labelledby="SocialLinks-title"
        className="SocialLinks__Nav-hNVsnJ jfaEKQ"
      >
        <FormattedMessage
          defaultMessage="Also find me on:"
          id="portfolio.social_links.header"
          values={Object {}}
        >
          <h1
            aria-hidden="false"
            hidden={true}
            id="SocialLinks-title"
          >
            Also find me on:
          </h1>
        </FormattedMessage>
        <StyledUL
          innerRef={[Function]}
        >
          <ul
            className="StyledUL-ePeKRx QkFKd"
          >
            <StyledLI
              data-tip="freeCodeCamp"
              key="https://www.freecodecamp.org/mensae"
            >
              <li
                className="StyledLI-jkNRrE hBftMa"
                data-tip="freeCodeCamp"
              >
                <ExternalLink
                  color="#465a65"
                  description="freeCodeCamp"
                  faIcon="free-code-camp"
                  href="https://www.freecodecamp.org/mensae"
                >
                  <StyledA
                    color="#465a65"
                    href="https://www.freecodecamp.org/mensae"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <a
                      className="StyledA-cLuoha bEATWP"
                      color="#465a65"
                      href="https://www.freecodecamp.org/mensae"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <IconFreeCodeCamp>
                        <SVGIcon
                          viewBox="0 0 576 512"
                        >
                          <svg
                            aria-hidden="true"
                            className="SVGIcon-ixpDyI gIDnTS"
                            focusable="false"
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M69.3 144.5c-41 68.5-36.4 163 1 227C92.5 409.7 120 423.9 120 438c0 6.8-6 13-12.8 13C87.7 451 8 375.5 8 253.2c0-111.5 78-186 97.1-186 6 0 14.9 4.8 14.9 11.1 0 12.7-28.3 28.6-50.7 66.2zm195.8 213.8c4.5 1.8 12.3 5.2 12.3-1.2 0-2.7-2.2-2.9-4.3-3.6-8.5-3.4-14-7.7-19.1-15.2-8.2-12.1-10.1-24.2-10.1-38.6 0-32.1 44.2-37.9 44.2-70 0-12.3-7.7-15.9-7.7-19.3 0-2.2.7-2.2 2.9-2.2 8 0 19.1 13.3 22.5 19.8 2.2 4.6 2.4 6 2.4 11.1 0 7-.7 14.2-.7 21.3 0 27 31.9 19.8 31.9 6.8 0-6-3.6-11.6-3.6-17.4 0-.7 0-1.2.7-1.2 3.4 0 9.4 7.7 11.1 10.1 5.8 8.9 8.5 20.8 8.5 31.4 0 32.4-29.5 49-29.5 56 0 1 2.9 7.7 12.1 1.9 29.7-15.1 53.1-47.6 53.1-89.8 0-33.6-8.7-57.7-32.1-82.6-3.9-4.1-16.4-16.9-22.5-16.9-8.2 0 7.2 18.6 7.2 31.2 0 7.2-4.8 12.3-12.3 12.3-11.6 0-14.5-25.4-15.9-33.3-5.8-33.8-12.8-58.2-46.4-74.1-10.4-5-36.5-11.8-36.5-2.2 0 2.4 2.7 4.1 4.6 5.1 9.2 5.6 19.6 21.4 19.6 38.2 0 46.1-57.7 88.2-57.7 136.2-.2 40.3 28.1 72.6 65.3 86.2zM470.4 67c-6 0-14.4 6.5-14.4 12.6 0 8.7 12.1 19.6 17.6 25.4 81.6 85.1 78.6 214.3 17.6 291-7 8.9-35.3 35.3-35.3 43.5 0 5.1 8.2 11.4 13.2 11.4 25.4 0 98.8-80.8 98.8-185.7C568 145.9 491.8 67 470.4 67zm-42.3 323.1H167c-9.4 0-15.5 7.5-15.5 16.4 0 8.5 7 15.5 15.5 15.5h261.1c9.4 0 11.9-7.5 11.9-16.4 0-8.5-3.5-15.5-11.9-15.5z"
                            />
                          </svg>
                        </SVGIcon>
                      </IconFreeCodeCamp>
                      <Description>
                        <span
                          className="Description-gMEclz ducwaF"
                        >
                          freeCodeCamp
                        </span>
                      </Description>
                    </a>
                  </StyledA>
                </ExternalLink>
              </li>
            </StyledLI>
            <StyledLI
              data-tip="GitHub"
              key="https://github.com/Mensae"
            >
              <li
                className="StyledLI-jkNRrE hBftMa"
                data-tip="GitHub"
              >
                <ExternalLink
                  color="#465a65"
                  description="GitHub"
                  faIcon="github"
                  href="https://github.com/Mensae"
                >
                  <StyledA
                    color="#465a65"
                    href="https://github.com/Mensae"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <a
                      className="StyledA-cLuoha bEATWP"
                      color="#465a65"
                      href="https://github.com/Mensae"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <IconGithub>
                        <SVGIcon
                          viewBox="0 0 496 512"
                        >
                          <svg
                            aria-hidden="true"
                            className="SVGIcon-ixpDyI gIDnTS"
                            focusable="false"
                            viewBox="0 0 496 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                            />
                          </svg>
                        </SVGIcon>
                      </IconGithub>
                      <Description>
                        <span
                          className="Description-gMEclz ducwaF"
                        >
                          GitHub
                        </span>
                      </Description>
                    </a>
                  </StyledA>
                </ExternalLink>
              </li>
            </StyledLI>
            <StyledLI
              data-tip="LinkedIn"
              key="https://nl.linkedin.com/in/jmeester"
            >
              <li
                className="StyledLI-jkNRrE hBftMa"
                data-tip="LinkedIn"
              >
                <ExternalLink
                  color="#465a65"
                  description="LinkedIn"
                  faIcon="linkedin"
                  href="https://nl.linkedin.com/in/jmeester"
                >
                  <StyledA
                    color="#465a65"
                    href="https://nl.linkedin.com/in/jmeester"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <a
                      className="StyledA-cLuoha bEATWP"
                      color="#465a65"
                      href="https://nl.linkedin.com/in/jmeester"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <IconLinkedin>
                        <SVGIcon
                          viewBox="0 0 448 512"
                        >
                          <svg
                            aria-hidden="true"
                            className="SVGIcon-ixpDyI gIDnTS"
                            focusable="false"
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                            />
                          </svg>
                        </SVGIcon>
                      </IconLinkedin>
                      <Description>
                        <span
                          className="Description-gMEclz ducwaF"
                        >
                          LinkedIn
                        </span>
                      </Description>
                    </a>
                  </StyledA>
                </ExternalLink>
              </li>
            </StyledLI>
            <StyledLI
              data-tip="Codepen"
              key="https://codepen.io/jmeester"
            >
              <li
                className="StyledLI-jkNRrE hBftMa"
                data-tip="Codepen"
              >
                <ExternalLink
                  color="#465a65"
                  description="Codepen"
                  faIcon="codepen"
                  href="https://codepen.io/jmeester"
                >
                  <StyledA
                    color="#465a65"
                    href="https://codepen.io/jmeester"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <a
                      className="StyledA-cLuoha bEATWP"
                      color="#465a65"
                      href="https://codepen.io/jmeester"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <IconCodepen>
                        <SVGIcon
                          viewBox="0 0 512 512"
                        >
                          <svg
                            aria-hidden="true"
                            className="SVGIcon-ixpDyI gIDnTS"
                            focusable="false"
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M502.285 159.704l-234-156c-7.987-4.915-16.511-4.96-24.571 0l-234 156C3.714 163.703 0 170.847 0 177.989v155.999c0 7.143 3.714 14.286 9.715 18.286l234 156.022c7.987 4.915 16.511 4.96 24.571 0l234-156.022c6-3.999 9.715-11.143 9.715-18.286V177.989c-.001-7.142-3.715-14.286-9.716-18.285zM278 63.131l172.286 114.858-76.857 51.429L278 165.703V63.131zm-44 0v102.572l-95.429 63.715-76.857-51.429L234 63.131zM44 219.132l55.143 36.857L44 292.846v-73.714zm190 229.715L61.714 333.989l76.857-51.429L234 346.275v102.572zm22-140.858l-77.715-52 77.715-52 77.715 52-77.715 52zm22 140.858V346.275l95.429-63.715 76.857 51.429L278 448.847zm190-156.001l-55.143-36.857L468 219.132v73.714z"
                            />
                          </svg>
                        </SVGIcon>
                      </IconCodepen>
                      <Description>
                        <span
                          className="Description-gMEclz ducwaF"
                        >
                          Codepen
                        </span>
                      </Description>
                    </a>
                  </StyledA>
                </ExternalLink>
              </li>
            </StyledLI>
          </ul>
        </StyledUL>
        <ReactTooltip
          insecure={true}
          resizeHide={true}
          wrapper="div"
        >
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          />
        </ReactTooltip>
      </nav>
    </SocialLinks__Nav>
  </SocialLinks>
</IntlProvider>
`;
